spec/models/migration_engine_spec.rb

```diff
@@ -435,4 +381,54 @@
       expect(engine.send(:sort_migrations, [[sle12], [sle12], [sle12]])).to eq([[sle12]])
     end
   end
+
+  describe '#add_migration_extras' do
+    subject(:method_call) { engine.send(:add_migration_extras, [[sle15]]) }
+
+    let!(:sle12sp1) do
+      create :product, :with_mirrored_repositories, :cloned, :activated,
+        system: system, from: sle12, name: 'sle12sp1',
+        version: '12.1', predecessors: [sle12]
+    end
+    let!(:sle15) do
+      create :product, :with_mirrored_repositories, :cloned, from: sle12sp1,
+        name: 'sle15', version: '15', predecessors: [sle12sp1]
+    end
+    let(:system) { FactoryGirl.create(:system, :with_activated_base_product) }
+    let(:installed_products) { [sle12sp1] }
+
+    context 'recommended modules do not get added automatically to the migration target' do
+      let!(:sle15_recommended_module) do
+        recommended_module = create(:product, :module, :with_mirrored_repositories,
+          name: 'sle15module', version: '15', base_products: [sle15])
+
+        ProductsExtensionsAssociation.find_by(
+          product: sle15,
+          extension: recommended_module,
+          root_product: sle15
+        ).update!(recommended: true)
+
+        recommended_module
+      end
+
+      it { is_expected.to contain_exactly([sle15, sle15_recommended_module]) }
+    end
+
+    context "modules with a 'migration_extra' flag do not get added automatically to the migration target" do
+      let!(:sle15_extra_module) do
+        extra_module = create(:product, :module, :with_mirrored_repositories,
+          name: 'sle15module', version: '15', base_products: [sle15])
+
+        ProductsExtensionsAssociation.find_by(
+          product: sle15,
+          extension: extra_module,
+          root_product: sle15
+        ).update!(migration_extra: true)
+
+        extra_module
+      end
+
+      it { is_expected.to contain_exactly([sle15, sle15_extra_module]) }
```

Comment:
the context name (`...do _not_ get added...`) and the test do not match

Reply:
fixed